org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/AbstractXbaseCompiler.java

```diff
@@ -35,10 +35,12 @@
 import org.eclipse.xtext.common.types.util.Primitives.Primitive;
 import org.eclipse.xtext.util.Strings;
 import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+import org.eclipse.xtext.xbase.XBinaryOperation;
```

Comment:
@cdietrich The package rename revealed more NPEs with broken models.

I added more safeguards below which you may want to double-check.